src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -34,12 +31,13 @@ public override IStep GetExtensionPostJobStep(IExecutionContext jobContext)
         public override string GetRootedPath(IExecutionContext context, string path)
         {
             string rootedPath = null;
+            TryGetRepositoryInfo(context, "self", out RepositoryInfo repoInfo);
 
-            if (SourceProvider != null &&
-                Repository != null &&
-                StringUtil.ConvertToBoolean(Repository.Properties.Get<string>("__AZP_READY")))
+            if (repoInfo.SourceProvider != null &&
```

Comment:
There is a null `repoInfo` check below but missing here.

Reply:
Yep, this object is never null, I will remove the checks.